src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -211,7 +215,12 @@ public boolean removePerson(ReadOnlyPerson key) throws PersonNotFoundException {
             removeContactPhoto(photoPath);
         }
 
-        return persons.remove(key);
+        if (persons.remove(key)) {
+            EventsCenter.getInstance().post(new PersonHasBeenDeletedEvent(key));
```

Comment:
Only raise the event on a successful deletion

Reply:
Raising event is on an AddressBook level as oppposed to a Model level.

It is much more clearer that you raise an event at the exact moment a **successful** deletion occurs as opposed to raising an event in all methods that involve a deletion of a person.
Similar to updating a person (during edit tag or delete tag)